tf2_ros/include/tf2_ros/buffer_interface.h

```diff
@@ -216,6 +216,21 @@ class BufferInterface
     return out;
   }
 
+  template<class T>
+  T & transform(
```

Comment:
I wonder for this one if the overload here makes sense. It's slightly more understandable to have transform_averaged() or something like that which explicitly lets you know that it's doing averaging. But maybe that implementation detail is obvious from the extra parameter required. Mostly thinking out loud here. And making sure that we're not boxing ourselves into the API, but I don't think that there's something else on the horizion with which this will collide.

Reply:
`transform_average` sounds good too. Open to change this too